### PR TITLE
Edit

### PR DESCRIPTION
**Pull request by ffasitella**
### Summary

- Added a view to edit an existing movie
- Added get and post methods to MoviesController

### Trello card

[https://trello.com/c/YLPWQXAW/7-edici%C3%B3n](https://trello.com/c/YLPWQXAW/7-edici%C3%B3n)

### CODE REVIEW

File: src/Views/Movies/Edit.cshtml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+            <div class="col-md-10">
+                <input asp-for="Price" class="form-control" />
+            </div>
+        </div>
+        <div class="form-group">
+            <div class="col-md-10">
+                <input type="submit" value="Edit" class="btn btn-primary"/>
+            </div>
+        </div>
+</form>
```
Comment (gzamudio) [inline src/Views/Movies/Edit.cshtml]: add empty line

File: src/Controllers/MoviesController.cs [hunk 2]
```diff
@@ -33,10 +33,39 @@ public IActionResult Create(MovieViewModel mvm)
...
+            if (ModelState.IsValid)
+            {
+                var movie = UnitOfWork.Movies.Get(mvm.Id);
+                movie.Price = mvm.Price;
+                movie.ReleaseDate = mvm.ReleaseDate;
+                movie.Title = mvm.Title;
+                movie.Genre = mvm.Genre;
+                UnitOfWork.Complete();
+            }
+            return RedirectToAction("Index", "Movies");
```
Comment (gzamudio) [inline src/Controllers/MoviesController.cs]: mmm. i think that if model state is invalid, we should not redirect to index.
if is invalid we must show the errors in the edit view.

File: src/Controllers/MoviesController.cs [hunk 3]
```diff
@@ -33,10 +33,39 @@ public IActionResult Create(MovieViewModel mvm)
...
         }
 
         [HttpGet("Index")]
         public IActionResult Index() => View(UnitOfWork.Movies.GetAll().Select(m => new MovieViewModel { Id = m.Id, ReleaseDate = m.ReleaseDate, Title = m.Title, Genre = m.Genre, Price = m.Price } ).ToList());
+    
+        [HttpGet("Edit")]
+        public IActionResult Edit(int Id)
+        {
+            var movie = UnitOfWork.Movies.Get(Id);
+            if (movie != null)
```
Comment (gzamudio) [inline src/Controllers/MoviesController.cs:43]: here we can use:
`if (movie != null)`
`     return View(new MovieViewModel { . . . } );`
`else`
`    return NotFound()`

**Review by gzamudio — CHANGES_REQUESTED** [review]

File: src/Controllers/MoviesController.cs [hunk 4]
```diff
@@ -33,10 +33,39 @@ public IActionResult Create(MovieViewModel mvm)
...
+            {
+                return View(new MovieViewModel { Id = movie.Id, Title = movie.Title, ReleaseDate = movie.ReleaseDate, Genre = movie.Genre, Price = movie.Price } );
+            }
+            else
+            {
+                return NotFound();
+            }
+        }
+
+        [HttpPost("Edit")]
```
Comment (gzamudio) [inline src/Controllers/MoviesController.cs]: also, i think that the method should be PUT instead of POST.

**Review by gzamudio — CHANGES_REQUESTED** [review]

**Review by gzamudio — APPROVED** [review]

**Review by mar7in91Wolox — APPROVED** [review]

**Pull request merged**